openhexa/toolbox/dhis2/dhis2.py

```diff
@@ -139,23 +153,35 @@ def datasets(self) -> List[dict]:
         for page in self.client.api.get_paged(
             "dataSets",
             params={
-                "fields": "id,name,dataSetElements,indicators,organisationUnits",
+                "fields": fields,
                 "pageSize": 10,
             },
         ):
+            fields_list = fields.split(",")
```

Comment:
`fields_list` is reset before looping  through the `dataSets`, we then potentially remove elements from it in the first iteration. So it means that it will only be applied to the first dataSet of each page ? Is that what we want ?

Reply:
Good catch, thanks